### PR TITLE
Add a list-languages option to joern-scan and joern-parse

### PR DESCRIPTION
**Pull request by johannescoetzee**
# Notes
Adds a `list-languages` option to list available languages with `joern-scan` and `joern-parse`, for example
```
❯ ./joern-scan --list-languages
Available languages (case insensitive):
- golang
- fuzzy_test_lang
- csharp
- java
- php
- c
- kotlin
- ghidra
- javascript
- python
- llvm
- newc
- javasrc
```

### CODE REVIEW

File: joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala [hunk 1]
```diff
@@ -85,12 +111,13 @@ object JoernParse extends App {
...
                           enhanceOnly: Boolean = false,
-                          language: String = "c")
+                          language: String = "",
+                          listLanguages: Boolean = false)
 
   def parseConfig(parserArgs: List[String]): Either[String, ParserConfig] =
     new scopt.OptionParser[ParserConfig]("joern-parse") {
       arg[String]("input")
-        .required()
+        .optional()
```
Comment (ursachec) [inline joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala:120]: why should this be optional?

Comment (johannescoetzee) [reply joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala:120]: It should be possible to run `joern-parse --list-languages` without providing an input path. If the `--list-languages` option isn't specified, there's a check later to make sure that a valid input path has been provided.

File: joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala [hunk 2]
```diff
@@ -39,6 +51,21 @@ object JoernParse extends App {
     }
   }
 
+  def checkInputPath(config: ParserConfig): Either[String, Unit] = {
+    if (config.inputPath == "" || !File(config.inputPath).exists) {
+      Left("Invalid input path provided")
```
Comment (ursachec) [inline joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala:56]:
```suggestion
      Left("Input path does not exist at `" + config.inputPath  + "`, exiting.")
```

File: joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala (hunk 1 above)
Comment (ursachec) [reply joern-cli/src/main/scala/io/shiftleft/joern/JoernParse.scala:121]: ah, ok, makes sense :+1:

**Review by ursachec — APPROVED** [review]

**Pull request merged**